Tools/TestWebKitAPI/Tests/WebKitCocoa/PermissionsAPI.mm

```diff
@@ -102,4 +124,44 @@ static void urlEncodeIfNeeded(uint8_t byte, StringBuilder& buffer)
     EXPECT_FALSE(didReceiveQueryPermission);
 }
 
+TEST(PermissionsAPI, PermissionMayHaveChanged)
+{
+    [TestProtocol registerWithScheme:@"https"];
+    auto webView = adoptNS([[TestWKWebView alloc] initWithFrame:CGRectMake(0, 0, 320, 500)]);
+
+    wasPermissionStatusChangedToGranted = false;
+    wasPermissionStatusChangedToPrompt = false;
+
+    [webView performAfterReceivingMessage:@"granted" action:^{
+        wasPermissionStatusChangedToGranted = true;
+    }];
+
+    [webView performAfterReceivingMessage:@"prompt" action:^{
+        wasPermissionStatusChangedToPrompt = true;
+    }];
+
+    auto delegate = adoptNS([[PermissionChangedTestAPIUIDelegate  alloc] init]);
+    [webView setUIDelegate:delegate.get()];
+
+    NSString *script = @"navigator.permissions.query({ name : 'notifications' }).then((permissionStatus) => {"
```

Comment:
Please use other permission (e.g. geolocation) as notifications is only enabled on macOS.